platforms/core-configuration/configuration-problems-base/src/main/kotlin/org/gradle/internal/configuration/problems/CommonReport.kt

```diff
@@ -229,8 +229,10 @@ class CommonReport(
                 val reportDir = outputDirectory.resolve(reportHash())
                 val reportFile = reportDir.resolve("$reportFileName.html")
                 if (!reportFile.exists()) {
-                    require(reportDir.mkdirs()) {
-                        "Could not create $reportFileName directory '$reportDir'"
+                    if (!reportDir.exists()) {
+                        require(reportDir.mkdirs()) {
+                            "Could not create $reportFileName directory '$reportDir'"
+                        }
```

Comment:
Slightly unrelated change. This handles the case where the user manually deletes the report file but keeps the container directory. Then, the next time the report generation is invoked, the build would fail because mkdirs()` returns false.